### DISCUSSION TITLE
Error compile oqtane server

### DISCUSSION THREAD

**Discussion by pnicosia80** [oqtane/oqtane.framework#3201] [Q&A] [answered]
Hi,
I tried to upgrade to 4.0.2 and the browser closes and oqtane doesn't update.
Also if I go to compile oqtane server I get the following error in visual studio. How can I fix?

Conflicting assets with the same target path '_framework/blazor.boot.json'. For assets 'Identity: C:\Sviluppo\Pietro\oqtane.framework\Oqtane.Client\bin\Debug\net7.0\wwwroot\_framework\blazor.boot.json, SourceType: Project, SourceId: Oqtane.Client, ContentRoot: C:\Sviluppo\Pietro\oqtane.framework\Oqtane.Client\bin\Debug\net7.0\wwwroot\, BasePath: /, RelativePath: _framework/blazor.boot.json, AssetKind: Build, AssetMode: All, AssetRole: Primary, RelatedAsset: , AssetTraitName: BlazorWebAssemblyResource, AssetTraitValue: manifest, CopyToOutputDirectory: PreserveNewest, CopyToPublishDirectory: Never, OriginalItemSpec: obj\Debug\net7.0\blazor.boot.json' and 'Identity: C:\Sviluppo\Pietro\oqtane.framework\Oqtane.Server\wwwroot\_framework\blazor.boot.json, SourceType: Discovered, SourceId: Oqtane.Server, ContentRoot: C:\Sviluppo\Pietro\oqtane.framework\Oqtane.Server\wwwroot\, BasePath: _content/Oqtane.Server, RelativePath: _framework/blazor.boot.json, AssetKind: All, AssetMode: All, AssetRole: Primary, RelatedAsset: , AssetTraitName: , AssetTraitValue: , CopyToOutputDirectory: Never, CopyToPublishDirectory: PreserveNewest, OriginalItemSpec: wwwroot\_framework\blazor.boot.json' from different projects.	
Oqtane.Server	
C:\Program Files\dotnet\sdk\7.0.306\Sdks\Microsoft.NET.Sdk.Razor\targets\Microsoft.NET.Sdk.Razor.StaticWebAssets.targets	
391	

Thanks

**Comment by sbwalker** [comment]
Can you please clarify what you mean when you say "I tried to upgrade to 4.0.2"... did you download the 4.0.2 official release and unzip it over top of your existing release, or did you use the System Update option in the framework, or ???. There were no compilation errors in the 4.0.2 release - how are you getting the source code - from the Master branch in Github - or the Dev branch or ??? It is not possible to assist you without sufficient information.

**Reply by pnicosia80** [reply]
System Update option in Oqtane Administration

**Comment by pnicosia80** [comment]
I fixed my build problem.
The error was update by system update from oqtane settings. It had downloaded dll or code and it no longer compiled. Discarding the changes and downloading the oqtane update from git compiled everything.
Thanks for the support

**Reply by sbwalker** [reply, marked as answer]
@adrianus80 the System Update feature is intended for upgrading a standard installation - not a development installation with source code. I am glad you were able to resolve your problem.
